private/privatetransactionmanager/tx_manager.go

```diff
@@ -15,46 +15,50 @@ type PrivateTransactionManager struct {
 	c    *gocache.Cache
 }
 
-func (g *PrivateTransactionManager) Send(data []byte, from string, to []string) (out []byte, err error) {
-	out, err = g.node.SendPayload(data, from, to)
+func (g *PrivateTransactionManager) Send(data []byte, from string, to []string) (out common.EncryptedPayloadHash, err error) {
+	var b []byte
+	b, err = g.node.SendPayload(data, from, to)
 	if err != nil {
-		return nil, err
+		return common.EncryptedPayloadHash{}, err
 	}
-	g.c.Set(string(out), data, cache.DefaultExpiration)
-	return out, nil
+	g.c.Set(string(b), data, cache.DefaultExpiration)
+	out = common.BytesToEncryptedPayloadHash(b)
+	return
 }
 
-func (g *PrivateTransactionManager) StoreRaw(data []byte, from string) (out []byte, err error) {
-	out, err = g.node.StorePayload(data, from)
+func (g *PrivateTransactionManager) StoreRaw(data []byte, from string) (out common.EncryptedPayloadHash, err error) {
+	var b []byte
+	b, err = g.node.StorePayload(data, from)
 	if err != nil {
-		return nil, err
+		return common.EncryptedPayloadHash{}, err
 	}
-	g.c.Set(string(out), data, cache.DefaultExpiration)
+	g.c.Set(string(b), data, cache.DefaultExpiration)
+	out = common.BytesToEncryptedPayloadHash(b)
 	return out, nil
 }
 
-func (g *PrivateTransactionManager) SendSignedTx(data []byte, to []string) (out []byte, err error) {
-	out, err = g.node.SendSignedPayload(data, to)
+func (g *PrivateTransactionManager) SendSignedTx(txHash common.EncryptedPayloadHash, to []string) (out []byte, err error) {
+	out, err = g.node.SendSignedPayload(txHash.Bytes(), to)
 	if err != nil {
 		return nil, err
 	}
 	return out, nil
 }
 
-func (g *PrivateTransactionManager) Receive(data []byte) ([]byte, error) {
-	if len(data) == 0 {
-		return data, nil
+func (g *PrivateTransactionManager) Receive(txHash common.EncryptedPayloadHash) ([]byte, error) {
+	if common.EmptyEncryptedPayloadHash(txHash) {
```

Comment:
I'm not sure that returning txHash.Bytes() is right here... 
An empty transaction.data field is converted to a 0 filled array (which is somewhat different from the previous behaviour).